server/pulp/server/webservices/views/consumers.py

```diff
@@ -0,0 +1,1082 @@
+        :param repo_id: A repo ID.
+        :type repo_id: str
+        :param distributor_id: A distributor ID.
+        :type distributor_id: str
+
+        :return: Response representing the binding
+        :rtype: django.http.HttpResponse
+        """
+
+        manager = factory.consumer_bind_manager()
+        bind = manager.get_bind(consumer_id, repo_id, distributor_id)
+        serialized_bind = serialization.binding.serialize(bind)
+        return generate_json_response_with_pulp_encoder(serialized_bind)
+
+    @auth_required(authorization.DELETE)
+    @json_body_allow_empty
+    def delete(self, request, consumer_id, repo_id, distributor_id):
+        """
+        Delete a bind association between the specified
+        consumer and repo-distributor.
+
+        :param request: WSGI request object
+        :type request: django.core.handlers.wsgi.WSGIRequest
+        :param consumer_id: A consumer ID.
+        :type consumer_id: str
+        :param repo_id: A repo ID.
+        :type repo_id: str
+        :param distributor_id: A distributor ID.
+        :type distributor_id: str
+
+        :raises OperationPostponed: will dispatch a task if 'notify_agent' is set to True
+        :raises InvalidValue: if some parameters are invalid
+
+        :return: Response representing the deleted binding(in case 'notify agent' is set to False)
+        :rtype: django.http.HttpResponse
+        """
+
+        body = request.body_as_json
+        forced = body.get('force', False)
+        if not isinstance(forced, bool):
+            raise InvalidValue(['force'])
+        options = body.get('options', {})
+        if not isinstance(options, dict):
+            raise InvalidValue(['options'])
+        if forced:
+            call_report = consumer_task.force_unbind(consumer_id, repo_id, distributor_id, options)
+        else:
+            call_report = consumer_task.unbind(consumer_id, repo_id, distributor_id, options)
+
+        if call_report.spawned_tasks:
+            raise OperationPostponed(call_report)
+        else:
+            return generate_json_response_with_pulp_encoder(call_report.serialize())
+
+
+class ConsumerContentActionView(View):
+    """
+    Views for content manipulation on the consumer.
+    """
+
+    @auth_required(authorization.CREATE)
+    @json_body_required
+    def post(self, request, consumer_id, action):
+        """
+        Install/update/uninstall content unit/s on the consumer.
+
+        :param request: WSGI request object
+        :type request: django.core.handlers.wsgi.WSGIRequest
+        :param consumer_id: A consumer ID.
+        :type consumer_id: str
+        :param action: type of action to perform
+        :type action: str
+
+        :raises MissingResource: if consumer id does not exist
+        """
+
+        method = getattr(self, action, None)
+        if method:
+            try:
+                factory.consumer_manager().get_consumer(consumer_id)
+            except:
```

Comment:
Looks like the only exception is a MissingResource, so lets not leave a catch-all `except`